src/pytest_bdd/gherkin_parser.py

```diff
@@ -67,135 +67,171 @@ def from_dict(cls, data: dict[str, Any]) -> Self:
+        ]
+
+        # Return a new DataTable with transposed rows
+        return DataTable(_location=self._location, rows=transposed_cells)
+
+    def to_dict(self) -> dict[str, list[str]]:
+        # Ensure there are at least two rows: one for the header and one for the values
+        if len(self.rows) < 2:
+            raise ValueError("DataTable needs at least two rows: one for headers and one for values")
+
+        # Extract the header row and the value rows
+        header = [cell.value for cell in self.rows[0].cells]
+        values_rows = [[cell.value for cell in row.cells] for row in self.rows[1:]]
+
+        # Transpose values and map headers to columns
+        return {key: list(values) for key, values in zip(header, zip(*values_rows))}
+
 
 @dataclass
 class DocString:
     content: str
     delimiter: str
-    location: Location
+    _location: Location
 
     @classmethod
     def from_dict(cls, data: dict[str, Any]) -> Self:
         return cls(
             content=textwrap.dedent(data["content"]),
             delimiter=data["delimiter"],
-            location=Location.from_dict(data["location"]),
+            _location=Location.from_dict(data["location"]),
         )
 
 
 @dataclass
 class Step:
-    id: str
+    _id: str
+    _location: Location
     keyword: str
-    keywordType: str
-    location: Location
+    keyword_type: str
     text: str
-    dataTable: DataTable | None = None
-    docString: DocString | None = None
+    data_table: DataTable | None = None
+    doc_string: DocString | None = None
 
     @classmethod
     def from_dict(cls, data: dict[str, Any]) -> Self:
         return cls(
-            id=data["id"],
+            _id=data["id"],
+            _location=Location.from_dict(data["location"]),
             keyword=data["keyword"].strip(),
-            keywordType=data["keywordType"],
-            location=Location.from_dict(data["location"]),
+            keyword_type=data["keywordType"],
             text=data["text"],
-            dataTable=DataTable.from_dict(data["dataTable"]) if data.get("dataTable") else None,
-            docString=DocString.from_dict(data["docString"]) if data.get("docString") else None,
+            data_table=DataTable.from_dict(data["dataTable"]) if data.get("dataTable") else None,
+            doc_string=DocString.from_dict(data["docString"]) if data.get("docString") else None,
         )
 
 
 @dataclass
 class Tag:
-    id: str
-    location: Location
+    _id: str
+    _location: Location
     name: str
 
     @classmethod
     def from_dict(cls, data: dict[str, Any]) -> Self:
-        return cls(id=data["id"], location=Location.from_dict(data["location"]), name=data["name"])
+        return cls(_id=data["id"], _location=Location.from_dict(data["location"]), name=data["name"])
 
 
 @dataclass
 class Scenario:
-    id: str
+    _id: str
+    _location: Location
```

Comment:
Yes - sorry, it was because I was spotting what looked like private attributes exposed in the data table object being returned to users (the location and id are meaningless to the user).  Agreed it should be in a separate PR